src/ResourceTemplate/Parser.php

```diff
@@ -215,11 +215,15 @@ private static function parseError($path, $index, $reason)
      * Check if $literal is a valid segment literal. Segment literals may only contain numbers,
      * letters, and any of the following: .-~_
      *
-     * @param string $literal
+     * @param string|null $literal
```

Comment:
Do we really want to accept `null` here? It seems like anyone passing in `null` here is doing so erroneously. 

We could change the signature to `string $literal` without breaking backwards compatibility: string casting would be taking place in `preg_match` anyway.